burn_fix/src/main.rs

```diff
@@ -0,0 +1,85 @@
+use indexer_core::{clap, clap::Parser, db, db::update, db::Connection, db::tables::metadatas, prelude::*};
+use solana_client::{client_error::{ClientError, ClientErrorKind}, rpc_request::RpcError, rpc_client::RpcClient};
+use solana_program::pubkey::Pubkey;
+
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[clap(flatten)]
+    db: db::ConnectArgs,
+    
+    #[clap(env)]
+    solana_endpoint: String,
+}
+
+fn main() {
+    indexer_core::run(|| {
+        let Opts { db, solana_endpoint } = Opts::parse();
+
+        let client = RpcClient::new(&solana_endpoint);
+        debug!("USING SOLANA ENDPOINT: {:?}", &solana_endpoint);
+
+        let db::ConnectResult {
+            pool,
+            ty: _,
+            migrated: _,
+        } = db::connect(db, db::ConnectMode::Write { migrate: false })?;
+
+        let conn = pool.get()?;
+
+        // TODO: Check for index on burned_at
+        let total_count = metadatas::table.filter(metadatas::burned_at.is_null()).count().get_result::<i64>(&conn).context("could not get count")?;
+        debug!("TOTAL UNBURNED COUNT: {:?}", total_count);
+
+        let mut i = 0;
+        let limit = 1000;
+        while i < total_count {
+
+            let addressess = metadatas::table.filter(metadatas::burned_at.is_null())
+                .offset(i)
+                .limit(limit)
+                .order_by(metadatas::slot.asc())
+                .select(metadatas::address)
+                .load(&conn)
+                .context("could not load metadatas")?;
+
+            addressess.into_iter().try_for_each(|address: String| {
+                let key: Pubkey = address.parse().unwrap();
```

Comment:
unwrap() should be removed to avoid panic. You can use context() from anyhow

Reply:
We need to keep track of NFTs that have been processed so that if we start again we know where to start

Reply:
What do you think about updated_at optional timestamp to be added into metadatas. We can then process metadatas `where burned_at is null AND updated_at is null`